oneanddone/tests/functional/tests/test_task_assignment.py

```diff
@@ -0,0 +1,22 @@
+# This Source Code Form is subjectfrom django import forms to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from pages.home import HomePage
+
+
+class TestAvailableTasks():
+    @pytest.mark.nondestructive
+    def test_assign_tasks(self, base_url, selenium, nonrepeatable_assigned_task, task, new_user):
+        home_page = HomePage(selenium, base_url).open()
+        home_page.login(new_user)
+        available_tasks_page = home_page.click_available_tasks()
+
+        # Check if assignable task is found
+        home_page.search_for_task(task.name)
+        assert len(available_tasks_page.available_tasks)
+
+        home_page.search_for_task(nonrepeatable_assigned_task.name)
+        assert len(available_tasks_page.available_tasks) == 0
```

Comment:
There are really two separate tests here. I would call one something like `test_that_unassigned_task_is_available` and the other `test_that_assigned_task_is_not_available`. Could you also add to the assigned task test a login as the assigned user and an assert that the task is listed on their dashboard? Without this, the test could be a false positive as the test may not exist at all.